pandas/src/generate_code.py

```diff
@@ -1,12 +1,23 @@
+"""This file generates `generated.pyx` which is then included in `../algos.pyx`
+during building.  To regenerate `generated.pyx`, just run:
+
+    `python generate_code.py`.
+
+"""
+
 from __future__ import print_function
-# we only need to be able to run this file on 2.7
-# don't introduce a pandas/pandas.compat import
-# or we get a bootstrapping problem
-from StringIO import StringIO
+import os
+from pandas.compat import StringIO
```

Comment:
I didn't understand this warning here and [the original commit](https://github.com/pydata/pandas/commit/21dabea425e1ee872fcfcaa36ecde991eb8d3a27) had no more information about this.  I tested this on Py2 and Py3 and it seemed to work fine...